lib/AST/Type.cpp

```diff
@@ -2027,9 +2027,14 @@ static ForeignRepresentableKind
 getObjCObjectRepresentable(Type type, const DeclContext *dc) {
   // @objc metatypes are representable when their instance type is.
   if (auto metatype = type->getAs<AnyMetatypeType>()) {
+    auto instanceType = metatype->getInstanceType();
+
+    // Consider protocol metatypes as not Objective-C representable.
+    if (metatype->is<MetatypeType>() && instanceType->is<ProtocolType>())
+      return ForeignRepresentableKind::None;
```

Comment:
True, metatypes of existential types were already rejected, but  metatypes of protocol composition crashed the same way as protocol metatypes before.

Added test cases for both of them.